servant-server/src/Servant/Server/Internal.hs

```diff
@@ -181,10 +182,10 @@ methodRouterEmpty method action = LeafRouter route'
     route' request respond
       | pathIsEmpty request && allowedMethod method request = do
           runAction action respond $ \ () ->
-            succeedWith $ responseLBS noContent204 [] ""
+            Route $! responseLBS noContent204 [] ""
```

Comment:
Again, why the `$!`. Ok, here it has an effect. But if you want to say that `Route` should always have a WHNF-evaluated argument, you should make the constructor strict, not each call.